File: LeetCode-Cpp/JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int l = nums[0];
        for (int i = 0; i < nums.size() - 1; i++) {
            l = max(l, nums[i]);
            l--;
            if (l == -1) return false;
        }

        return true;
    }
};

File: LeetCode-Python/JumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        l = len(nums)

        if l==1:
            return True
        s = nums[0]

        for i in range(l-1):
            if nums[i] > s:
                s = nums[i]
            s = s - 1
            if s < 0 :
                return False
        return True